mc/watch.go

```diff
@@ -119,14 +115,24 @@ func (w *Watcher) Wait() {
 	w.wg.Wait()
 }
 
-// Join the watcher with client
-func (w *Watcher) Join(client Client, recursive bool) *probe.Error {
```

Comment:
You've removed the check for ClientWatcher, I added this because in theory it could be possible to have a Client not supporting a watcher.